### DISCUSSION TITLE
Idle Speculation

### DISCUSSION THREAD

**Discussion by RandomDSdevel** [ravynsoft/ravynos#193] [General]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I had a few thoughts/musings and questions:  

 - Some good resources, if you weren't aware of them, might be:  
   - Jonathan Levin's _macOS and *OS Internals, 2<sup>nd</sup> Edition_ (web site [here](http://newosxbook.com/)) (_Note:_  I've also mentioned this on PureDarwin's Discord server.)  
   - The ['hack-different/apple-knowledge' repository](https://github.com/hack-different/apple-knowledge) on GitHub.  
 - When it comes to drivers, I don't know how much the Hackintosh community's efforts in that space might help, if at all, but maybe you could adapt some driver code from [the Asahi Linux project](https://github.com/AsahiLinux) for part(s) of this?  (Though that could well be more help after somebody started working on trying to get airyxOS running on an ARM-based system, if at all.)  (I think I remember hearing somebody working on this in a at least one BSD, but I recall neither whom nor which one that was at the moment.)  
 - Regarding UI, I saw you considering relying on Qt 5 as an implementation detail, then noting the some of the problems you were having with attempting to do that; did you encounter any other issues besides those?  (I was also going to ask what Qt 5 has which Qt 6 still doesn't that makes it so you have to use the former instead of the latter, but I saw you already have the newer version on your radar.)  
 - Will airyxOS's GUI have any theming support?  Anybody who shares sentiment in line with the title of the MacRumors forum thread ['Why is Big Sur so ugly?'](https://forums.macrumors.com/threads/why-is-big-sur-so-ugly.2243081/) would undoubtedly appreciate at least a macOS Catalina 10.15.x theme.  (Edit:  Though having equivalents to Big Sur's and Monterey's accessibility options to reduce transparency, increase contrast, and show button shapes would help mitigate this somewhat/a little bit, but the option would still be quite nice.)  
 - On that note, similarly, other people might appreciate the ability to run alternative desktop environments on airyxOS, though I have at least some suspicion that making them compatible with existing system applications would likely take morre than a bit of work.  
 - I don't know how relevant this is here, but, for a non-trivial amount of time now, I've considered '`launchd`' a bit too monolithic and wondered what it might look like broken up into some more modular parts, if that'd be at all possible.  
 - Have you considered streaming any live coding sessions while tinkering on the project for those who might find interest in watching them either live or on-demand later via recording?  (I don't currently know if I'd personally be one of those people myself, but I suspect others might well be.)  
 - Some blog posts about 'adventures' you've had and/or amusing/curious findings both so far and in the future during project development might well also be interesting.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Regarding:  

> ⁝ (_Snipped…_)  
>
> - I don't know how relevant this is here, but, for a non-trivial amount of time now, I've considered '`launchd` ' a bit too monolithic and wondered what it might look like broken up into some more modular parts, if that'd be at all possible.  
>
> ⁝ (_Snipped…_)  

I've written some notes and initial thoughts on that up [here](https://github.com/RandomDSdevel/Developer-Project-Index/discussions/2) for the time being.

**Comment by mszoek** [comment]
There's some good thoughts here which I somehow missed when originally posted. Thanks for the pointers to resources!

Several people have brought up themes and UI appearance, so I think we will try to do something there. There is no way I'm going to stream myself coding - I won't subject any of you to that kind of torture :) Blog posts might be possible.

**Reply by RandomDSdevel** [reply]
> There's some good thoughts here which I somehow missed when originally posted. Thanks for the pointers to resources!  
>
> ⁝ (_Snipped…_)

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;You're very welcome.  

> …There is no way I'm going to stream myself coding - I won't subject any of you to that kind of torture :)  …

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Eh, I've seen stuff that's pretty/fairly informal and unstructured.  Your call, though, of course.  

> …Blog posts might be possible.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Looking forward to them, then.
